src/capi.rs

```diff
@@ -23,12 +24,18 @@ pub unsafe extern "C" fn microvmi_envlogger_init() {
     env_logger::init();
 }
 
+/// Entrypoint for libmicrovmi
+/// Initializes a specific driver, or all drivers compiled and returns the first one that succeeded
+///
+/// In case of error, init_error will be allocated with the underlying error message.
+/// The init_error pointer should be freed with rs_cstring_free()
 #[allow(clippy::missing_safety_doc)]
 #[no_mangle]
 pub unsafe extern "C" fn microvmi_init(
     domain_name: *const c_char,
     driver_type: *const DriverType,
     driver_init_option: *const DriverInitParamFFI,
+    init_error: *mut *const c_char,
```

Comment:
if the initialization fails, I don't see what the user can do except log and exit.
of course it would be better to give more meaningful info to the C world, what type of error is reported for example.

libvmi fills an enum to try to convey the cause of the failure.
https://github.com/libvmi/libvmi/blob/master/libvmi/libvmi.h#L93

libmicrovmi is already doing better by providing a precise error string, and not a generic one i'd say.

Reply:
f that's okay for you, you can resolve this conversation.